catalogs/catalogio.py

```diff
@@ -120,6 +120,16 @@ def get_control_by_id(self, control_id: str) -> dict:
         )
         return result_dict
 
+    def get_next_control_by_id(self, control_id: str):
+        search_collection = self.get_control_ids()
+        next = False
+        for item in search_collection:
+            if next is True:
+                return item
+            if item == control_id:
+                next = True
+        return ""
```

Comment:
I still think an error should be raised if the `control_id` isn't found in the `search_collection` list. I don't think there's anything at the view layer that checks this either, so there'd be no distinction between the end of the list and a non-existent control.

This would require a couple of things: 
- Raise an error (possibly a custom error so it's easier to handle downstream)
- Handle the error in the appropriate view(s) and return a 404 response with a clear message.
- Additional test for trying to access a non-existent control

Reply:
This should not raise an error, it should just end with an empty string being returned. If that is not what will happen, then that is unexpected.

Reply:
Just for a little more context; 
If you are on the last control, then this is the information to display the page for the last control. Which is why I say this should not be an error, cause then you could not complete the last control.

Reply:
> This should not raise an error, it should just end with an empty string being returned. If that is not what will happen, then that is unexpected.

I'm saying that an error _should_ be raised, not that there actually is one currently. I'm saying that the current ambiguity between the "next" control from a non-existent input `control_id` and the input `control_id` already being the last in the list needs to be addressed and handled.